### ISSUE TITLE
Nullable List Relationships

### ISSUE THREAD

**Issue by adley-kim** [neo4j/graphql#1258]
I'm having difficulty understanding the decision to remove support for nullable lists on relationship fields as of @neo4j/graphql@3.0.0. In the [v3.0.0 announcement post](https://neo4j.com/developer-blog/announcing-the-release-of-the-neo4j-graphql-library-3-0-0/), @darrellwarde explains why several variants of the following type definition are invalid.

```
type Movie {
    title: String!
    actors: [Actor] @relationship(type: "ACTED_IN", direction: IN)
}
```

What I don't understand is why `actors: [Actor]! @relationship(type: "ACTED_IN", direction: IN)` is no longer considered valid. A new movie may have been announced but hasn't cast any actors. This doesn't strike me as a fringe scenario but it would throw an error under the new constraints.

I'm working on a project where I anticipate a lot of sparse data, so I'm wondering if I need to change my approach to type definitions? If so, is there a recommended workaround for dealing with nullable list relationships?

**Comment by darrellwarde** [comment]
> `actors: [Actor]! @relationship(type: "ACTED_IN", direction: IN)`

For this example, the problem is as per my blog post - `Actor` being nullable means that conceptually, relationships with no terminating nodes would be possible:

![image](https://user-images.githubusercontent.com/8117355/163219491-ebeacf39-60ad-41d6-837f-1ffc3e7d8533.png)

> A new movie may have been announced but hasn't cast any actors. This doesn't strike me as a fringe scenario but it would throw an error under the new constraints.

A new movie being created with no actors is still perfectly valid under the new restrictions using the following type definitions:

```gql
type Movie {
    title: String!
    actors: [Actor!]! @relationship(type: "ACTED_IN", direction: IN)
}
```

Just because `actors` is non-nullable, it doesn't mean that it can't be an empty array, which is what the appropriate value would be in this scenario.

In summary, _nullability_ isn't really appropriate for lists - the list should instead be empty. I'm going to close this - happy to discuss but Discord is perhaps a better place for that: https://discord.gg/neo4j
